src/sigma.ts

```diff
@@ -560,50 +567,111 @@ export default class Sigma<GraphType extends Graph = Graph> extends TypedEventEm
   private bindGraphHandlers(): this {
     const graph = this.graph;
 
-    this.activeListeners.graphUpdate = () => {
-      this.scheduleRefresh();
+    this.activeListeners.eachNodeAttributesUpdatedGraphUpdate = (e: { hints?: { attributes?: string[] } }) => {
+      const updatedFields = e.hints?.attributes;
```

Comment:
Those hints are completely optional. I guess we need to consider everything might have changed if the hint was not provided to avoid frustration? It might be what the current code does.